tasks.c

```diff
@@ -3426,7 +3429,9 @@ void vTaskMissedYield( void )
  * void prvIdleTask( void *pvParameters );
  *
  */
-static portTASK_FUNCTION( prvIdleTask, pvParameters )
+static void prvIdleTask( void * pvParameters ) __attribute__( ( __noreturn__ ) );
```

Comment:
This change would probably not work for all compilers? If so, we need to remove it.

Reply:
Added `portNORETURN` to all of the known GNU compiler ports and the `FreeRTOS.h` file if not defined.  This will solve the issue of not being supported on some compilers.  Note for C23 - noreturn is supported.

For IAR it is `__noreturn` based on: https://wwwfiles.iar.com/arm/webic/doc/EWARM_DevelopmentGuide.ENU.pdf
For CCS did not look into it.

But would suggest you specify compiler specific defines in one location at the top of each `portable/GCC` `portable/IAR`, etc. so that they are defined once and included in each of the portable.h files under the respective compiler.  That way you are guaranteed not to miss one in one or some of the ports (which is the case for some of these values at the moment).

But out of scope of this PR.